src/openfermion/transforms/_conversion_test.py

```diff
@@ -507,3 +513,374 @@ def test_get_fermion_operator_majorana_operator():
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=False)
+
+        for i in range(self.molecule.n_qubits):
+            n_op = FermionOperator(((i, 1), (i, 0)))
+            sum_n_op += n_op
+
+            sparse_n_op = get_number_preserving_sparse_operator(
+                n_op,
+                self.molecule.n_qubits,
+                self.molecule.n_electrons,
+                spin_preserving=False)
+
+            sum_sparse_n_op += sparse_n_op
+
+            expectation = reference.dot(sparse_n_op.dot(reference))
+
+            if i < self.molecule.n_electrons:
+                assert expectation == 1.0
+            else:
+                assert expectation == 0.0
+
+        convert_after_adding = get_number_preserving_sparse_operator(
+            sum_n_op,
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=False)
+
+        assert scipy.sparse.linalg.norm(convert_after_adding - sum_sparse_n_op) < 1E-9
+
+        assert reference.dot(sum_sparse_n_op.dot(reference)) - self.molecule.n_electrons < 1E-9
+
+
+    def test_hf_energy(self):
+        hamiltonian_fop = get_fermion_operator(self.molecular_hamiltonian)
+
+        sparse_ham = get_number_preserving_sparse_operator(
+            hamiltonian_fop,
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=True)
+
+        reference = self.get_reference_state(
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=True)
+
+        sparse_hf_energy = reference.dot(sparse_ham.dot(reference))
+
+        assert numpy.linalg.norm(sparse_hf_energy - self.molecule.hf_energy) < 1E-9
+
+
+    def test_one_body_hf_energy(self):
+        one_body_part = self.molecular_hamiltonian
+        one_body_part.two_body_tensor = numpy.zeros_like(one_body_part.two_body_tensor)
+
+        one_body_fop = get_fermion_operator(one_body_part)
+        one_body_regular_sparse_op = get_sparse_operator(one_body_fop)
+
+        make_hf_fop = FermionOperator(((3, 1), (2, 1), (1, 1), (0, 1)))
+        make_hf_sparse_op = get_sparse_operator(make_hf_fop, n_qubits=12)
+
+        hf_state = numpy.zeros((2**12))
+        hf_state[0] = 1.0
+        hf_state = make_hf_sparse_op.dot(hf_state)
+
+        regular_sparse_hf_energy = (hf_state @ (one_body_regular_sparse_op.dot(hf_state))).real
+
+        one_body_sparse_op = get_number_preserving_sparse_operator(
+            one_body_fop,
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=True)
+
+        reference = self.get_reference_state(
+            self.molecule.n_qubits,
+            self.molecule.n_electrons,
+            spin_preserving=True)
+
+        sparse_hf_energy = reference @ (one_body_sparse_op.dot(reference))
+
+        assert numpy.linalg.norm(sparse_hf_energy - regular_sparse_hf_energy) < 1E-9
```

Comment:
Add a test where you construct the sparse operator in two different ways: using your new method, and using `get_sparse_operator` combined with `jw_number_restrict_operator` and `jw_sz_restrict_operator`. Verify that the entire spectrums match.

Reply:
This is a really great test!